### DISCUSSION TITLE
AirPlay immediately turns off

### DISCUSSION THREAD

**Discussion by cYCL157** [rern/rAudio#447] [Bug Reports]
I seem to be having an issue with AirPlay right now, when to cast an audiobook and noticed the device was not available.  I could see that it was off and when I turn it on, it immediately turns off.  Any suggestions on how I can troubleshoot to fix or provide meaningful information??

**Comment by RchrdCambl** [comment]
I had the same problem. In my case the configuration file /etc/shairport-sync.conf had somehow become corrupted. To troubleshoot, view the airplay log by clicking on the AirPlay label next to the button used to enable AirPlay. 

Here is my log:
# pacman -Q shairport-sync; systemctl status shairport-sync

shairport-sync 3.3.8-1
× shairport-sync.service - Shairport Sync - AirPlay Audio Receiver
     Loaded: loaded (/usr/lib/systemd/system/shairport-sync.service; enabled; vendor preset: disabled)
    Drop-In: /etc/systemd/system/shairport-sync.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Sat 2021-08-14 20:24:16 CDT; 11s ago
    Process: 7246 ExecStart=/usr/bin/taskset -c 3 /usr/bin/shairport-sync (code=exited, status=1/FAILURE)
   Main PID: 7246 (code=exited, status=1/FAILURE)
        CPU: 54ms

Aug 14 20:24:16 rAudio systemd[1]: Started Shairport Sync - AirPlay Audio Receiver.
Aug 14 20:24:16 rAudio shairport-sync[7246]: fatal error: Line 8 of the configuration file "/etc/shairport-sync.conf":
                                             syntax error
Aug 14 20:24:16 rAudio systemd[1]: shairport-sync.service: Main process exited, code=exited, status=1/FAILURE
Aug 14 20:24:16 rAudio systemd[1]: shairport-sync.service: Failed with result 'exit-code'.

### Note the line:
Aug 14 20:24:16 rAudio shairport-sync[7246]: fatal error: Line 8 of the configuration file "/etc/shairport-sync.conf":
                                             syntax error

### Use secure shell (ssh) from a terminal window to log in to the rAudio machine and examine the configuration file as follows:
ssh root@raudio.local
root@raudio.local's password: ros
rAudio:! # cat /etc/shairport-sync.conf

### The result should look something like:
general = {
	name = "rAudio"
}
sessioncontrol = {
	run_this_before_play_begins = "/usr/bin/sudo /srv/http/bash/shairport.sh";
	run_this_after_play_ends = "/usr/bin/sudo /srv/http/bash/shairport.sh stop";
}
alsa = {
	output_device = "hw:1";
	mixer_control_name = "( not available )";
}
alsa = {
	output_device = "hw:1";
	mixer_control_name = "( not available )";
}

### In my case, the first alsa = { line looked like this in the vi editor:
general = {
	name = "rAudio"
}
sessioncontrol = {
	run_this_before_play_begins = "/usr/bin/sudo /srv/http/bash/shairport.sh
";
  	run_this_after_play_ends = "/usr/bin/sudo /srv/http/bash/shairport.sh st
op";
}
\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200
\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200
\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200
\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200alsa
 = {
    	output_device = "hw:1";
	mixer_control_name = "( not available )";
}
alsa = {
        output_device = "hw:1";
	mixer_control_name = "( not available )";
}

### Use your favorite editor to delete the first _alsa_ stanza:
alsa = {
	output_device = "hw:1";
	mixer_control_name = "( not available )";
}

Reboot your rAudio machine.
